JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/derivation/DerivationClient.java

```diff
@@ -123,6 +123,12 @@ public List<String> bulkCreateDerivations(List<List<String>> entitiesList, List<
 		// add timestamp to each derivation
 		this.sparqlClient.addTimeInstance(derivations);
 
+		// validate derivations if more than one derivations are marked up
+		// this is to prevent the potential circular dependencies in the markup added but not detected at creation
+		if (derivations.size() > 1) {
+			validateDerivations();
```

Comment:
This method validates all the Derivation instances in the KG, if you want to validate them, you should only validate the ones created in the current method

Reply:
maybe it's fine